src/pages/home/report/ReportActionItemParentAction.tsx

```diff
@@ -1,91 +1,105 @@
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
     const {isSmallScreenWidth} = useWindowDimensions();
-    const parentReportAction = parentReportActions?.[`${report?.parentReportActionID ?? ''}`] ?? null;
+    const report = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`];
+    const allAncestors = ReportUtils.getAllAncestorReportActions(report, shouldHideThreadDividerLine, allReports, allReportActions);
 
-    // In case of transaction threads, we do not want to render the parent report action.
-    if (ReportActionsUtils.isTransactionThread(parentReportAction)) {
-        return null;
-    }
     return (
-        <OfflineWithFeedback
-            shouldDisableOpacity={Boolean(parentReportAction?.pendingAction ?? false)}
-            pendingAction={report?.pendingFields?.addWorkspaceRoom ?? report?.pendingFields?.createChat}
-            errors={report?.errorFields?.addWorkspaceRoom ?? report?.errorFields?.createChat}
-            errorRowStyles={[styles.ml10, styles.mr2]}
-            onClose={() => Report.navigateToConciergeChatAndDeleteReport(report?.reportID ?? '0')}
-        >
-            <View style={StyleUtils.getReportWelcomeContainerStyle(isSmallScreenWidth)}>
+        <>
+            <View style={[StyleUtils.getReportWelcomeContainerStyle(isSmallScreenWidth), styles.justifyContentEnd]}>
                 <AnimatedEmptyStateBackground />
                 <View style={[styles.p5, StyleUtils.getReportWelcomeTopMarginStyle(isSmallScreenWidth)]} />
-                {parentReportAction && (
-                    <ReportActionItem
-                        // @ts-expect-error TODO: Remove the comment after ReportActionItem is migrated to TypeScript.
-                        report={report}
-                        action={parentReportAction}
-                        displayAsGroup={false}
-                        isMostRecentIOUReportAction={false}
-                        shouldDisplayNewMarker={shouldDisplayNewMarker}
-                        index={index}
-                    />
-                )}
+                {allAncestors.map((ancestor) => (
+                    <OfflineWithFeedback
+                        shouldDisableOpacity={Boolean(ancestor.reportAction?.pendingAction)}
+                        pendingAction={ancestor.report?.pendingFields?.addWorkspaceRoom ?? ancestor.report?.pendingFields?.createChat}
+                        errors={ancestor.report?.errorFields?.addWorkspaceRoom ?? ancestor.report?.errorFields?.createChat}
+                        errorRowStyles={[styles.ml10, styles.mr2]}
+                        onClose={() => Report.navigateToConciergeChatAndDeleteReport(ancestor.report.reportID)}
+                    >
+                        <ReportActionItem
+                            key={ancestor.reportAction.reportActionID}
```

Comment:
It's still showing the warning. We might move it to the above component to clear the warning.